src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferTypedStringBuilderAppendOverloads.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>CA1830: Prefer strongly-typed StringBuilder.Append overloads.</summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class PreferTypedStringBuilderAppendOverloads : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1830";
```

Comment:
Once you merge latest, you would now need to bump up the range end/next available ID: https://github.com/dotnet/roslyn-analyzers/blob/78dd233d304b93b5d430b0a96fafe83f8344d932/src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt#L15